### ISSUE TITLE
Runner returns success when check failed

### ISSUE THREAD

**Issue by sebastiaanluca** [sebastianfeldmann/phpbu#54]
The point here is that using `$result = $this->runner->run($configuration);` the following should return `false` when a __check failed__: `$result->wasSuccessful()`. But in my test case, it returns true as in _success_. While it should return that the entire backup failed because the check failed, right?

Debugged it and boils down to this (I could be totally wrong though):

1. Added a check to a backup
2. Check fails and https://github.com/sebastianfeldmann/phpbu/blob/master/src/Result.php#L285 is called. This adds a failed check and also fails the backup.
3. This only updates the backup, not the __result__ which is used to determine a successful backup run! This is done here: https://github.com/sebastianfeldmann/phpbu/blob/master/src/Result.php#L133 where the value being used (`backupsFailed`) is only updated when a backup itself failed (https://github.com/sebastianfeldmann/phpbu/blob/master/src/Result.php#L237). The `checksFailed` value is not being used anywhere except in the verbose CLI logging, I think?

In short, shouldn't `wasSuccessful` in the result class also check for failed checks, syncs, … instead of just backups?

**Comment by sebastianfeldmann** [comment]
Hi,
`$result->wasSuccessful()` currently focuses on pure backup execution. I admit this is somehow confusing.
The method you could use right now is `$result->allOk()`
Sadly the current version has a bug and is missing to validate some values.
From todays point of view this whole skip, fail, success mixing makes my head hurt and I should simplify this in the future.

This
https://github.com/sebastianfeldmann/phpbu/commit/46b8eb20e80572e758647cd980a3bb70b96352e4
fixes the `$result->allOk()` bug

But I think you are right, a failing check should fail the backup as well.
There's actually a test called
```
public function testBackupFailedOnFailedCheck()
```

**Comment by sebastianfeldmann** [comment]
Fixed in version 3.0.1, once again thanks for reporting this.

**Comment by sebastianfeldmann** [comment]
To make sure everything (backup, check, crypt, sync and cleanup) worked you still have to use `$result->allOk()`
The method `$result->wasSuccessful()` is still limited to the backup and it's checks.

**Comment by sebastiaanluca** [comment]
You're too good! Thanks man :)
